app/javascript/components/dashboard/configuration-profiles/edit/Agents.jsx

```diff
@@ -0,0 +1,296 @@
+import React, { Fragment, useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  setCurrentConfigurationProfile,
+  setEditCPErrors,
+  setSavingCP,
+} from "../../../../actions/configurationProfiles";
+import updateCP from "../../../../services/updateCP";
+import ConfirmDialog from "../../../shared/ConfirmDialog";
+
+const Agents = (props) => {
+  const { currentDSOIndex } = props;
+  const [agentsData, setAgentsData] = useState(props.agentsData);
+  const [currentAgentIndex, setCurrentAgentIndex] = useState(0);
+  const configurationProfile = useSelector((state) => state.currentCP);
+  const [agentFullname, setAgentFullname] = useState(
+    agentsData[currentAgentIndex].fullname
```

Comment:
An exception will occur if the `agentsData` array is empty.

Reply:
Covered.